### ISSUE TITLE
[FRONTEND] Implement edit functionality on profile page

### ISSUE THREAD

**Issue by bcho892** [UoaWDCC/uasc-web#337]
**Is your feature request related to a problem? Please describe.**

Blocked by #336 

You will need to pass in edit handlers to the component created in #327 

[A mutation hook](https://tanstack.com/query/v4/docs/framework/react/reference/useMutation) also needs to be created that calls the `edit-self` endpoint

**BEFORE MERGING**

- [ ] Storybooks created where possible
- [ ] Tested with [nock](https://tanstack.com/query/v3/docs/framework/react/guides/testing) if data fetching
- [ ] Tests written for critical interactions
- [ ] PR Reviewed (For non-trivial changes)
- [ ] Changes tested after rebasing on master or merging in master (_hint_: `git fetch origin master:master`, then `git rebase master` or `git merge master`)
- [ ] All required PR checks passing

**Comment by bcho892** [comment]
Already done in #509 and #520
